service/lib/dinstaller/manager.rb

```diff
@@ -176,6 +179,21 @@ def probe_steps
       status_manager.change(Status::Probed.new)
     end
 
+    # Performs required steps after installing the system
+    #
+    # For now, this only unmounts the installed system and copies installation logs. Note that YaST
+    # performs many more steps like copying configuration files, creating snapshots, etc. Adding
+    # more features to D-Installer could require to recover some of that YaST logic.
+    def finish_installation
+      progress.init_minor_steps(2, "Copying logs")
```

Comment:
NP: I'd say here are 3 minor steps, namely: "Copying logs", "Unmounting target system", "Target system correctly unmounted". Although the latest one is like a _success message_ instead of a minor step.

Reply:
> NP: I'd say here are 3 minor steps, namely: "Copying logs", "Unmounting target system", "Target system correctly unmounted". Although the latest one is like a _success message_ instead of a minor step.
> 
> So I suggest either, to use a 3 anyway 

Never mind, I was wrong https://github.com/yast/d-installer/blob/913186d48ef6d6fae6c8e4c97ecc7c943421a8fd/service/lib/dinstaller/progress.rb#L43 since 2 means 3: 0, 1, and 2 :confused: 


> or to remove the latest _next_minor_step_ call.

But the latest step still looking more like feedback about the 2 step that a step itself.

Anyway, PR LGTM.

Reply:
I don't like the current API. I would prefer something like:

~~~ruby
progress = Progress.new

progress.init_steps(3)
progress.step("Doing step 1")
progress.step("Doing step 2")
progress.init_sub_steps(2)
progress.sub_step("Doing step 2.1")
progress.sub_step("Doing step 2.2")
progress.finish_sub_steps
progress.step("Doing step 3")
progress.finish
~~~

Reply:
> I don't like the current API. I would prefer something like:
> ...

I guess we can think about it and make needed changes if we found/decide/agree on better/different approach.